tests/main/snap-quota-groups/task.yaml

```diff
@@ -0,0 +1,76 @@
+    exit 0
+  fi
+
+  echo "Create a group with a snap in it"
+  snap quota group-one --max-memory=100MB go-example-webserver
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should also still be active"
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+active"
+
+  echo "The systemd slice should have one process in it now"
+  cgroupProcsFile="/sys/fs/cgroup/memory/$sliceName/snap.go-example-webserver.webserver.service/cgroup.procs"
+  wc -l < "$cgroupProcsFile" | MATCH 1
+  SERVER_PID=$(cat "$cgroupProcsFile")
+
+  echo "And that process is the main PID for the example webserver"
+  systemctl show --property=MainPID snap.go-example-webserver.webserver.service | MATCH "MainPID=$SERVER_PID"
+
+  echo "And the service is in the Control Group for the slice"
+  test "$(systemctl show --property=ControlGroup snap.go-example-webserver.webserver.service)" = "ControlGroup=/$sliceName/snap.go-example-webserver.webserver.service"
+
+  # TODO: check that systemctl says the memory usage is the same as "snap quota group-one"
+
+  echo "Removing the quota will stop the slice and the service will be restarted"
+  snap remove-quota group-one
+  systemctl show --property=MainPID snap.go-example-webserver.webserver.service | NOMATCH "MainPID=$SERVER_PID"
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+active"
+
+  echo "And the service is not in a slice anymore"
+  systemctl show --property=ControlGroup snap.go-example-webserver.webserver.service | NOMATCH "/$sliceName/snap.go-example-webserver.webserver.service"
+
+  echo "Creating a quota with a very small memory limit results in the service being unable to start"
+  snap quota too-small --max-memory=1MB go-example-webserver
+
+  echo "The systemd slice should be active"
+  sliceName="snap.$(systemd-escape --path too-small).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "But the service is not running after a short amount of time"
+  sleep 10 # meh probably unnecessarily long
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+inactive"
+  systemctl show --property=ExecMainStatus snap.go-example-webserver.webserver.service | MATCH ExecMainStatus=9
+
+  echo "And after removing the quota group, the snap is restarted and active again"
+  snap remove-quota too-small
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+active"
+
+  echo "Removing a snap ensures that the snap is not in the quota group anymore"
+  snap quota group-three --max-memory=100MB go-example-webserver
+  snap quota group-three | yaml2json | jq -r '.snaps | .[]' | MATCH go-example-webserver
```

Comment:
Maybe `snap quotas | MATCH ...` instead?

Reply:
Ah, scratch that, we need individual snaps so what you did is correct and the only way.